math/mathcore/inc/Fit/LogLikelihoodFCN.h

```diff
@@ -118,20 +118,20 @@ class LogLikelihoodFCN : public BasicFCN<DerivFunType,ModelFunType,UnBinData>  {
    virtual unsigned int NFitPoints() const { return fNEffPoints; }
 
    /// i-th likelihood contribution and its gradient
-   virtual double DataElement(const double * x, unsigned int i, double * g, double *  h = nullptr, bool fullHessian = false) const {
+   double DataElement(const double * x, unsigned int i, double * g, double *  h = nullptr, bool fullHessian = false) const override {
       if (i==0) this->UpdateNCalls();
       return FitUtil::Evaluate<T>::EvalPdf(BaseFCN::ModelFunction(), BaseFCN::Data(), x, i, g, h, BaseFCN::IsAGradFCN(), fullHessian);
    }
 
    // need to be virtual to be instantiated
-   virtual void Gradient(const double *x, double *g) const {
+   void Gradient(const double *x, double *g) const {
```

Comment:
Amazing work @ferdymercury!

I only don't feel so comfortable with these functions where you remove the `virtual` keyword without replacing it with `override`. I understand the dilemma, but I think this deserves special attention and maybe some refactoring, which should better be done in a separate PR.

Can you maybe restrict this PR to the "simple" changes where you either just add `override`, or replace `virtual` with `override` because it was in a clearly derived class to begin with?